.github/workflows/checkstatus.yml

```diff
@@ -4,26 +4,41 @@ on:
   status:
 
 jobs:
+
   Check-statuses:
     runs-on: ubuntu-latest
-
     steps:
+
     - name: Check status
-      run: |
-        curl -s --url https://api.github.com/repos/${{ github.repository }}/commits/${{ github.event.sha }}/status \
-        --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' | head -n 2 | tail -n 1 | grep "success"
+      id: check
+      run: >-
+        curl -fsSL
+        -H 'Accept: application/vnd.github.v3+json'
+        -H 'authorization: Bearer ${{ github.token }}'
+        https://api.github.com/repos/${{ github.repository }}/commits/${{ github.event.sha }}/status
+        | head -n 2
+        | tail -n 1
+        | grep "success"
+        || echo '::set-output name=skip::true'
+
     - name: Set up Python
+      if: steps.check.outputs.skip != 'true'
```

Comment:
Is no way to wrap all the remaining steps in the if?

Reply:
Unfortunately, I don't think so: https://stackoverflow.com/questions/60589373/how-to-force-to-exit-in-github-actions-step

It would be possible to define two jobs (instead of one job with multiple steps). However, then we'd need to pass some parameters from one job to another. I think that'd be too contrived...